x-pack/metricbeat/module/ibmmq/status/_meta/testdata/ibmmq-status.9.1.4.0-r1-amd64.plain-expected.json

```diff
@@ -0,0 +1,407 @@
+[
+    {
+        "event": {
+            "dataset": "ibmmq.status",
+            "duration": 115000,
+            "module": "ibmmq"
+        },
+        "metricset": {
+            "name": "status",
+            "period": 10000
+        },
+        "prometheus": {
+            "labels": {
+                "handler": "prometheus",
+                "instance": "127.0.0.1:60576",
+                "job": "ibmmq",
+                "quantile": "0.9"
+            },
+            "metrics": {
+                "http_request_duration_microseconds": 11240.8,
+                "http_request_size_bytes": 541,
+                "http_response_size_bytes": 17385
+            }
+        },
+        "service": {
+            "address": "127.0.0.1:55555",
+            "type": "ibmmq"
+        }
+    },
+    {
+        "event": {
+            "dataset": "ibmmq.status",
+            "duration": 115000,
+            "module": "ibmmq"
+        },
+        "metricset": {
+            "name": "status",
+            "period": 10000
+        },
+        "prometheus": {
+            "labels": {
+                "instance": "127.0.0.1:60576",
+                "job": "ibmmq",
+                "quantile": "0.25"
+            },
+            "metrics": {
+                "go_gc_duration_seconds": 0.0002584
+            }
+        },
+        "service": {
+            "address": "127.0.0.1:55555",
+            "type": "ibmmq"
+        }
+    },
+    {
+        "event": {
+            "dataset": "ibmmq.status",
+            "duration": 115000,
+            "module": "ibmmq"
+        },
+        "metricset": {
+            "name": "status",
+            "period": 10000
+        },
+        "prometheus": {
+            "labels": {
+                "instance": "127.0.0.1:60576",
+                "job": "ibmmq"
+            },
+            "metrics": {
+                "go_gc_duration_seconds_count": 4,
```

Comment:
Do we really want to store all these fields?

Reply:
@ChrsMark  Could you please elaborate on that? Are you afraid of too many metrics being collected?

Reply:
Yeap, excuse my brevity. What I mean is that I'm not sure if we have to store metrics like `go_gc_duration_seconds_count`. There are many of this kind in all prometheus exporters. My personal point of view is that we should not collect them here and stick to only the IBM specific ones. Wondering if we could achieve that since the implementation is with a light module 🤔

Reply:
Well, I like this idea, but... not really sure if it should be in scope of this task. What do you think about creating an issue for that and solving as a separate feature?

Reply:
yeap it is for sure, but I think we should be very opinionated when we store fields to avoid  having ES filled with not-so-important data. +1 on raising this concern in a different issue.

Reply:
Issue: https://github.com/elastic/beats/issues/15493

Reply:
@exekias @sorantis 
do you think it's blocking for delivery of IBM MQ?

Reply:
Isn't it possible to deliver an ingest pipeline that is doing the job on Elasticsearch side?

Reply:
AFAIK an ingest pipeline can be used to filter/process log data with grok expressions (field oriented). I'm not sure if we want to hack it to reject metrics...

Reply:
[remove-processor](https://www.elastic.co/guide/en/elasticsearch/reference/current/remove-processor.html )=> removes fields. No hack. Very easy. The only thing I don't know is weather pipeline creation via beat setup only works with Filebeat.

Reply:
"Remove fields" requires to provide a defined list of fields, no regular expressions.With next releases of IBM MQ, we may have some fields sneaked.

I'm thing rather about a script processors to reject all metrics without "ibmmq_" prefix.

Reply:
Hey @mtojek! https://github.com/elastic/beats/blob/a1e3539c2d6befbc0c3842e21d79ed53ba6947c5/x-pack/metricbeat/module/cockroachdb/status/_meta/docs.asciidoc#L15 might be useful here too, not checked in deep though.

Reply:
I have just pushed a basic filtering using Javascript processor. It filters prometheus metrics based on metrics name prefix.